update to llama.cpp b5688

arnej27959 opened the pull request:
- extract updated glue code to server.hpp and utils.cpp
- adapt native code in jllama.cpp to track API changes
- update tags and adapt CMakeLists.txt

note: All the code in utils.hpp and server.hpp is just lifted directly from llama.cpp, so I don't really have a good understanding of all the changes that happened there.  But I've looked closely at the changes that needed to be done in jllama.cpp to track the API changes, and done a bit of testing of the resulting binary.

longer-term it would be better to get the server code in llama.cpp split so it these pieces can be used as a library directly; we are considering writing a different server wrapper with protobuf/rpc replacing JSON/http, which could use the same library.

arnej27959 committed:
update to llama.cpp b5688

- extract updated glue code to server.hpp and utils.cpp
- adapt native code in jllama.cpp to track API changes
- update tags and adapt CMakeLists.txt

arnej27959 committed:
fix my mistake regarding '--reranking' option

arnej27959 committed:
let LLAMA_CURL default to OFF for now

kherud commented:
Awesome work :+1:  though it'll take some time for me to review.

> longer-term it would be better to get the server code in llama.cpp split so it these pieces can be used as a library directly; 

Yeah, if you can get the llama.cpp team to do this, it would make things much easier!

> we are considering writing a different server wrapper with protobuf/rpc replacing JSON/http, which could use the same library.

I'm a big fan of protobuf/rpc, so I'll look forward to that!

arnej27959 committed:
track code in oaicompat_chat_params_parse

comment says:
// if the assistant message appears at the end of list, we do not add end-of-turn token

so it seems the changed output should be is expected.

arnej27959 commented:
any progress on looking at these changes?

bernardladenthin commented:
The project no longer seems to be maintained. I would like to make some changes, add Java 1.8 compatibility, upgrade the library, and include existing and open pull requests. This PR has been integrated: https://github.com/bernardladenthin/java-llama.cpp/
